superfly/flyctl
# Allow fly logs to read --machine flag

### Change Summary

**What and Why:**
Allow `fly logs` to read the `--machine` flag or `--instance` flag to identify the machine id to filter logs for.

**How:**
1. Include a `--machine` [flag](https://github.com/superfly/flyctl/pull/3820/files#diff-2b3f3552f1349c8d16af492e7cf1f26a0cb8aa6a01faceeea4b521835833b4adR59) for the `fly logs` command
2. Make sure to update `--instance` flag as an[ alias](https://github.com/superfly/flyctl/pull/3820/files#diff-2b3f3552f1349c8d16af492e7cf1f26a0cb8aa6a01faceeea4b521835833b4adR56) to `--machine` flag.
3. Add a boolean attribute [UseAliasShortHand](https://github.com/superfly/flyctl/pull/3820/files#diff-cd4ae47be757809e391f0b84469d6a2e9a6299fb56bacc30164a89119ff5fb4eR110) in `flag.String` struct. And [read this in `makeAlias()`](https://github.com/superfly/flyctl/pull/3820/files#diff-cd4ae47be757809e391f0b84469d6a2e9a6299fb56bacc30164a89119ff5fb4eR40) to determine whether to include aliases' first characters as shorthand
3. Include a [UseAliasShortHand](https://github.com/superfly/flyctl/pull/3820/files#diff-2b3f3552f1349c8d16af492e7cf1f26a0cb8aa6a01faceeea4b521835833b4adR57) in the `machine` flag in order to allow the inclusion of its aliases( namely: --instance )' shorthands( namely: -i )

**Related to:**

https://flyio.discourse.team/t/approach-to-inconsistency-in-flags-found-in-flyctl-part-3-instance-vs-machine/6682/2
---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team

### How To Test Manually:
- Primary function: Run `fly logs --app <appName> --machine <machineId>`
- Other flag combinations to make sure --instance, and --machine, works with proper behavior,  please refer to [preflight test, here!](https://github.com/superfly/flyctl/pull/3820/files#diff-24f9f64b6f7b7bab981deb41053356c0217b42a43122efcfbacfce9a5c4cb123R30-R47)

### [Preflight Tests](https://github.com/superfly/flyctl/blob/master/CONTRIBUTING.md#preflight)!
- For running preflight test, checking that `fly logs` work in general with the subject flags: `make preflight-test T=TestFlyLogsMachineFlagBehavior`
